cpp/libc.cc

```diff
@@ -82,7 +82,7 @@ List<Str*>* regex_match(Str* pattern, Str* str) {
   int match;
   const char* s0 = str0.Get();
   regmatch_t* pmatch = (regmatch_t*)malloc(sizeof(regmatch_t) * outlen);
-  if (match = (regexec(&pat, s0, outlen, pmatch, 0) == 0)) {
+  if ( (match = (regexec(&pat, s0, outlen, pmatch, 0) == 0)) ) {
```

Comment:
Can you make this a separate assignment and then `if (match == 0)` ?  See `native/libc.c`, I did that there to silence this warning

(that is the Python version of this function, which is unfortunately duplicated)